In [ ]:
import pandas as pd

ratings=pd.read_csv("./data/ramen-ratings.csv")

#ratings.Brand #ratings["Brand"]

#ratings.Stars[80:950]  get the stars column ranging from index 80 to 950




# Index-based selection:
    # first part is the ROWS and the other is the COLUMN number
    # (:) means all rows , (4) is the column number
    


#ratings.iloc[:,4] 
#ratings.iloc[1:4,3]

#ratings.iloc[-5:] #Last 5 rows





# Label-based selection:


ratings.loc[50:1000,['Brand','Variety','Style']]


Brand                                            Variety Style
50           Ottogi           Ppushu Ppushu Noodle Snack Honey Butter   Pack
51         Maruchan                            Gotsumori Chanpon Ramen  Bowl
52            Prima  Juzz's Mee Creamy Chicken Flavour (Export Vers...  Pack
53    Uni-President                                    Mushroom Flavor  Pack
54    Samyang Foods                                 Jjamppong Big Bowl  Bowl
...             ...                                                ...   ...
996            E-mi                             Mi Perencah Asam Pedas  Pack
997            J.J.                                 Snek Mi Perisa BBQ  Pack
998        Mi E-Zee                        Perisa Kari / Curry Flavour  Pack
999          United  3 Con Tom Instant Noodles Thailand Sour Shrimp...  Pack
1000          Emart             Dare You! Habanero Ramen (New Edition)  Pack

[951 rows x 3 columns]

#Choosing between loc and iloc
When choosing or transitioning between loc and iloc, there is one "gotcha" worth keeping in mind, which is that the two methods use slightly different indexing schemes.

iloc uses the Python stdlib indexing scheme, where the first element of the range is included and the last one excluded. So 0:10 will select entries 0,...,9. loc, meanwhile, indexes inclusively. So 0:10 will select entries 0,...,10.

Why the change? Remember that loc can index any stdlib type: strings, for example. If we have a DataFrame with index values Apples, ..., Potatoes, ..., and we want to select "all the alphabetical fruit choices between Apples and Potatoes", then it's a lot more convenient to index df.loc['Apples':'Potatoes'] than it is to index something like df.loc['Apples', 'Potatoet'] (t coming after s in the alphabet).

This is particularly confusing when the DataFrame index is a simple numerical list, e.g. 0,...,1000. In this case df.iloc[0:1000] will return 1000 entries, while df.loc[0:1000] return 1001 of them! To get 1000 elements using loc, you will need to go one lower and ask for df.loc[0:999].

Otherwise, the semantics of using loc are the same as those for iloc.

In [ ]:
ratings.set_index("Brand") # set a column into the index of the result of the dataset

Review #                                            Variety  \
Brand                                                                         
New Touch           2580                          T's Restaurant Tantanmen    
Just Way            2579  Noodles Spicy Hot Sesame Spicy Hot Sesame Guan...   
Nissin              2578                      Cup Noodles Chicken Vegetable   
Wei Lih             2577                      GGE Ramen Snack Tomato Flavor   
Ching's Secret      2576                                    Singapore Curry   
...                  ...                                                ...   
Vifon                  5  Hu Tiu Nam Vang ["Phnom Penh" style] Asian Sty...   
Wai Wai                4                     Oriental Style Instant Noodles   
Wai Wai                3                                     Tom Yum Shrimp   
Wai Wai                2                               Tom Yum Chili Flavor   
Westbrae               1                                         Miso Ramen   

               Style   Country Stars Top Ten  
Brand                                         
New Touch        Cup     Japan  3.75     NaN  
Just Way        Pack    Taiwan     1     NaN  
Nissin           Cup       USA  2.25     NaN  
Wei Lih         Pack    Taiwan  2.75     NaN  
Ching's Secret  Pack     India  3.75     NaN  
...              ...       ...   ...     ...  
Vifon           Bowl   Vietnam   3.5     NaN  
Wai Wai         Pack  Thailand     1     NaN  
Wai Wai         Pack  Thailand     2     NaN  
Wai Wai         Pack  Thailand     2     NaN  
Westbrae        Pack       USA   0.5     NaN  

[2580 rows x 6 columns]

In [ ]:
#ratings.loc[(ratings.Brand=="Wai Wai")& (ratings.Style=="cup") |(ratings.Style=="Bowl")]
#ratings.loc[(ratings.Brand=="Wai Wai")& ratings.Style.isin(["cup","Bowl"])]



ratings.loc[ratings.Variety.notnull()]

Review #    Brand                               Variety Style   Country  \
861      1719  Wai Wai  Grilled Pork Flavour Instant Noodles  Bowl  Thailand   

    Stars Top Ten  
861   2.5     NaN